app/views/additional_payments/claims/ineligible.html.erb

```diff
@@ -1,5 +1,4 @@
-<% require "./lib/ineligibility_reason_checker.rb" %>
-<% reason = IneligibilityReasonChecker.new(current_claim).reason %>
+<% reason = IneligibilityReasonChecker.new(current_claim:, journey_session:).reason %>
```

Comment:
One thing that might we worth considering is passing the `ClaimSessionShim` into this class, and replacing all the calls to current claim with that which will work with both the session and the claim. That's what I went for https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/CAPT-1650/itt-year-form/lib/ineligibility_reason_checker.rb

Reply:
I originally tried but ran into some issues (don't recall the failures), but i can have another play